### DISCUSSION TITLE
Getting FrameTreeSnapshot

### DISCUSSION THREAD

**Discussion by marlow-fawn** [bdaiinstitute/spot_ros2#450] [Q&A] [answered]
Hello,

We're trying to recreate the fetch demo with ROS2. We would like to use the PickObjectInImage message, but I'm having trouble figuring out how to get a FrameTreeSnapshot message, which is required for that message. Is the only option currently to manually populate one from tf? Or send an empty IK request and get the message that way?

**Comment by khughes-bdai** [comment, marked as answer]
Hello,

Unfortunately, there is currently not an easy way for getting the FrameTreeSnapshot from the ros2 driver. For our internal examples that use this message, we manually create a snapshot by iterating through the tf frames. It is possible that sending an empty IK request could also get you this info, but I have not tried this.
